sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/implementation/models/Path.java

```diff
@@ -318,12 +313,81 @@ public String getETag() {
 
     /**
      * Set the eTag property: The etag property.
-     *
+     * 
      * @param eTag the eTag value to set.
      * @return the Path object itself.
      */
     public Path setETag(String eTag) {
         this.eTag = eTag;
         return this;
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
+        jsonWriter.writeStartObject();
+        jsonWriter.writeStringField("name", this.name);
+        jsonWriter.writeBooleanField("isDirectory", this.isDirectory);
+        jsonWriter.writeStringField("lastModified", this.lastModified);
+        jsonWriter.writeNumberField("contentLength", this.contentLength);
+        jsonWriter.writeStringField("owner", this.owner);
+        jsonWriter.writeStringField("group", this.group);
+        jsonWriter.writeStringField("permissions", this.permissions);
+        jsonWriter.writeStringField("EncryptionScope", this.encryptionScope);
+        jsonWriter.writeStringField("creationTime", this.creationTime);
+        jsonWriter.writeStringField("expiryTime", this.expiryTime);
+        jsonWriter.writeStringField("EncryptionContext", this.encryptionContext);
+        jsonWriter.writeStringField("etag", this.eTag);
+        return jsonWriter.writeEndObject();
+    }
+
+    /**
+     * Reads an instance of Path from the JsonReader.
+     * 
+     * @param jsonReader The JsonReader being read.
+     * @return An instance of Path if the JsonReader was pointing to an instance of it, or null if it was pointing to
+     * JSON null.
+     * @throws IOException If an error occurs while reading the Path.
+     */
+    public static Path fromJson(JsonReader jsonReader) throws IOException {
+        return jsonReader.readObject(reader -> {
+            Path deserializedPath = new Path();
+            while (reader.nextToken() != JsonToken.END_OBJECT) {
+                String fieldName = reader.getFieldName();
+                reader.nextToken();
+
+                if ("name".equals(fieldName)) {
+                    deserializedPath.name = reader.getString();
+                } else if ("isDirectory".equals(fieldName)) {
+                    deserializedPath.isDirectory = reader.getNullable(JsonReader::getBoolean);
+                } else if ("lastModified".equals(fieldName)) {
+                    deserializedPath.lastModified = reader.getString();
+                } else if ("contentLength".equals(fieldName)) {
+                    deserializedPath.contentLength = reader.getNullable(JsonReader::getLong);
```

Comment:
Appears the service returns this as a "String" so we'll need a transform here. I'd say to cover all bases it'd be good to do something more general like

```java
JsonToken token = reader.currentToken();
if (token == JsonToken.STRING) {
    deserializedPath.contentLength = Long.parseLong(reader.getString());
} else if (token == JsonToken.NUMBER) {
    deserializedPath.contentLength = reader.getLong();
} else if (token == JsonToken.NULL) {
    deserializedPath.contentLength = null;
} else {
    throw new IllegalStateException("Invalid token, expected one of STRING, NUMBER, or NULL. Was " + token);
}